src/utils.rs

```diff
@@ -33,9 +35,24 @@ pub fn escape_html(input: &str) -> String {
     output
 }
 
+pub(crate) fn render_to_string<F, E>(render: F) -> Result<String, Error>
+where
+    F: FnOnce(&mut Vec<u8>) -> Result<(), E>,
+    Error: From<E>,
+{
+    let mut buffer = Vec::new();
+    render(&mut buffer).map_err(Error::from)?;
+    Ok(buffer_to_string(buffer))
+}
+
+pub(crate) fn buffer_to_string(buffer: Vec<u8>) -> String {
+    String::from_utf8(buffer).expect("Render only UTF-8 bytes")
```

Comment:
let's handle that error as well rather than `expect`

Reply:
Okay, but this will introduce a new error type (and at the moment this error should never happen as we only write UTF8 bytes)

Reply:
It's probably ok, we have non-exhaustive error enums.

Reply:
It should be ok now